star-sys/STAR/source/MmapArray.h

```diff
@@ -0,0 +1,22 @@
+#ifndef MMAPARRAY_DEF
+#define MMAPARRAY_DEF
+
+#include "IncludeDefine.h"
+
+class MmapArray {
+    private:
+        int fd;
+
+    public:
+        uint file_mmap_length;
+        char *file_mmap;
+        char *prefix_mmap;
+        char *last_page_mmap;
```

Comment:
The 4 things you should be keeping track of are
1. The start of the mmapped region. `char *`.
2. The start of the array, which may be after the start of the mmapped region, if the stuff allocated before the file was intended to be less than the page size. `char *`.
3. The total length of the mmap region `size_t`.
4. The total length of the array `size_t`.

There is NEVER a good reason to use uint for anything.